src/braille.jl

```diff
@@ -8,37 +11,42 @@ Binary dither with algorithm `alg`, then print image in braille.
 
 ## Keyword arguments
   - `invert`: invert Unicode output, defaults to `false`
+  - `to_string`: return a string instead of printing to terminal. Defaults to `false`.
 
 All keyword arguments for binary dithering methods can be used.
 """
-function braille(img::GenericImage, alg::AbstractDither; invert=false, kwargs...)
+function braille(
+    img::GenericImage,
+    alg::AbstractDither;
+    invert::Bool=false,
+    to_string::Bool=false,
+    kwargs...,
+)
     d = dither(Bool, Gray.(img), alg; kwargs...)
-    invert && return brailleprint(.!d)
-    return brailleprint(d)
+    return braille(d; invert=invert, to_string=to_string)
 end
 
-"""
-    brailleprint(img)
+function braille(A::AbstractMatrix{Bool}; invert::Bool=false, to_string::Bool=false)
+    invert && (A .= .!A)
+    B = _braille_matrix(A)
+    str = join([join(r) for r in eachrow(B)], "\n")
```

Comment:
These two `join` seems only necessary for `to_string=true` case.
For `to_string=false` case, it might be sufficient to print it, e.g.,

```julia
if to_string
    str = ...
    return str
end
io = IOBuffer()
foreach(eachrow(B)) do line
```
    foreach(c->print(io, c), line)
    println(io)
end
print(String(take!(io)))
```

Reply:
I'm now using `String(take!(io))` to build the string, and then either returning it or printing it based on `to_string`.